client/app/scripts/superdesk-workspace/content/content.js

```diff
@@ -146,6 +148,16 @@
                     });
                 });
 
+                scope.$watch(function() {
+                    return scope.reloadTemplatesList;
```

Comment:
Just a minor comment: since we know the value of reloadTemplatesList can be set to true only at line 129 then maybe we don't need to create a watch, we can call the function to load directly.